src/gConsent/query/query.ts

```diff
@@ -0,0 +1,262 @@
+//
+// Copyright Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import LinkHeader from "http-link-header";
+
+import { handleErrorResponse } from "@inrupt/solid-client-errors";
+import type { DatasetWithId } from "@inrupt/solid-client-vc";
+import { verifiableCredentialToDataset } from "@inrupt/solid-client-vc";
+import type { UrlString } from "@inrupt/solid-client";
+
+/**
+ * Supported Access Credentials status
+ */
+export type CredentialStatus =
+  | "Pending"
+  | "Denied"
+  | "Granted"
+  | "Canceled"
+  | "Expired"
+  | "Active"
+  | "Revoked";
+
+/**
+ * Supported Access Credential types
+ */
+export type CredentialType =
+  | "SolidAccessRequest"
+  | "SolidAccessGrant"
+  | "SolidAccessDenial";
+
+/**
+ * Supported durations for Access Credential filtering.
+ */
+export const DURATION = {
+  ONE_DAY: "P1D",
+  ONE_WEEK: "P7D",
+  ONE_MONTH: "P1M",
+  THREE_MONTHS: "P3M",
+} as const;
+
+export type CredentialFilter = {
+  /**
+   * The Access Credential type (e.g. Access Request, Access Grant...)
+   */
+  type?: CredentialType;
+  /**
+   * The Access Credential status (e.g. Active, Revoked...)
+   */
+  status?: CredentialStatus;
+  /**
+   * WebID of the Agent who issued the Access Credential.
+   */
+  fromAgent?: UrlString;
+  /**
+   * WebID of the Agent who is the Access Credential recipient.
+   */
+  toAgent?: UrlString;
+  /**
+   * URL of the resource featured in the Access Credential.
+   */
+  resource?: UrlString;
+  /**
+   * URL of the Access Credential purpose.
+   */
+  purpose?: UrlString;
+  /**
+   * Interval (expressed using ISO 8601) during which the Credential was issued.
```

Comment:
```suggestion
   * Period (expressed using ISO 8601) during which the Credential was issued.
```
I think this is slightly better - this post suggests the differences: https://www.quora.com/What-is-the-difference-between-a-period-duration-and-interval